domain-management/src/main/java/org/jboss/as/domain/management/parsing/ManagementXml_5.java

```diff
@@ -2046,22 +2075,33 @@ private void writeServerIdentities(XMLExtendedStreamWriter writer, ModelNode rea
             }
 
             boolean hasProvider = ssl.hasDefined(KEYSTORE_PROVIDER) && !JKS.equalsIgnoreCase(ssl.require(KEYSTORE_PROVIDER).asString());
+            boolean hasCredential = ssl.hasDefined(KeystoreAttributes.KEYSTORE_PASSWORD_CREDENTIAL_REFERENCE.getName()) || ssl.hasDefined(KeystoreAttributes.KEY_PASSWORD_CREDENTIAL_REFERENCE.getName());
             if (hasProvider || ssl.hasDefined(KeystoreAttributes.KEYSTORE_PATH.getName())) {
-                writer.writeEmptyElement(Element.KEYSTORE.getLocalName());
+                if(hasCredential) {
+                    writer.writeStartElement(Element.KEYSTORE.getLocalName());
+                } else {
+                    writer.writeEmptyElement(Element.KEYSTORE.getLocalName());
+                }
                 KeystoreAttributes.KEYSTORE_PROVIDER.marshallAsAttribute(ssl, writer);
                 KeystoreAttributes.KEYSTORE_PATH.marshallAsAttribute(ssl, writer);
                 KeystoreAttributes.KEYSTORE_RELATIVE_TO.marshallAsAttribute(ssl, writer);
                 KeystoreAttributes.KEYSTORE_PASSWORD.marshallAsAttribute(ssl, writer);
                 KeystoreAttributes.ALIAS.marshallAsAttribute(ssl, writer);
                 KeystoreAttributes.KEY_PASSWORD.marshallAsAttribute(ssl, writer);
                 KeystoreAttributes.GENERATE_SELF_SIGNED_CERTIFICATE_HOST.marshallAsAttribute(ssl, writer);
+                KeystoreAttributes.KEYSTORE_PASSWORD_CREDENTIAL_REFERENCE.marshallAsElement(ssl, writer);
```

Comment:
I'm marshalling as element while the other attributes are marshalled as attribute. Changing the order will not respect the xml.